db/config.c

```diff
@@ -684,6 +685,85 @@ static int new_table_from_schema(struct dbenv *dbenv, char *tblname,
     return 0;
 }
 
+/*
+ * Toggles the 'modsnap' snapshot implementation on and off.
+ *
+ * switch_val: 1 if toggling on and 0 if toggling off.
+ */
+static void toggle_modsnap(int switch_val) {
+    gbl_use_modsnap_for_snapshot = switch_val;
+    gbl_modsnap_asof = switch_val;
+}
+
+/*
+ * Toggles the 'new' snapshot implementation on and off.
+ *
+ * switch_val: 1 if toggling on and 0 if toggling off.
+ */
+static void toggle_new_snapisol(int switch_val) {
+    gbl_new_snapisol = switch_val;
+    gbl_new_snapisol_asof = switch_val;
+    gbl_new_snapisol_logging = switch_val;
+}
+
+/*
+ * Prints a string representation of a `snap_impl_enum` type,
+ * or "UNKNOWN" if the value passed in is not a valid `snap_impl_enum`.
+ *
+ * impl: The value to output as a string.
+ */
+static const char *snap_impl_str(snap_impl_enum impl) {
+    switch (impl) {
+    case SNAP_IMPL_ORIG:
+        return "ORIGINAL";
+        break;
+    case SNAP_IMPL_NEW:
+        return "NEW";
+        break;
+    case SNAP_IMPL_MODSNAP:
+        return "MODSNAP";
+        break;
+    default:
+        return "UNKNOWN";
+        break;
+    }
+}
+
+/*
+ * Sets the snapshot implementation.
+ *
+ * impl: The implementation to be set.
+ */
+void set_snapshot_impl(snap_impl_enum impl) {
+    if (gbl_snap_impl == impl) {
+        return;
+    }
+
+    gbl_snap_impl = impl;
+    gbl_snap_impl == SNAP_IMPL_MODSNAP ? toggle_modsnap(1) : toggle_modsnap(0);
+    gbl_snap_impl == SNAP_IMPL_NEW ? toggle_new_snapisol(1) : toggle_new_snapisol(0);
+
+    logmsg(LOGMSG_USER, "%s: Switched snapshot implementation to '%s'\n",
+                        __func__, snap_impl_str(gbl_snap_impl));
```

Comment:
Is this log-level too chatty?

Reply:
Seems fine I think, wouldn't expect to be switching that often?